### ISSUE TITLE
Broken ssl module: TypeError: __init__() got an unexpected keyword argument 'server_hostname'

### ISSUE THREAD

**Issue by Glandos** [searx/searx#120]
This is a continuation of what I am experiencing in #92 .

Here is the complete stacktrace, for reference:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/var/www/search.example.com/searx/venv/local/lib/python2.7/site-packages/grequests.py", line 71, in send
    self.url, **merged_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 456, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/sessions.py", line 559, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/dist-packages/requests/adapters.py", line 327, in send
    timeout=timeout
  File "/usr/lib/python2.7/dist-packages/urllib3/connectionpool.py", line 516, in urlopen
    body=body, headers=headers)
  File "/usr/lib/python2.7/dist-packages/urllib3/connectionpool.py", line 304, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python2.7/dist-packages/urllib3/connectionpool.py", line 724, in _validate_conn
    conn.connect()
  File "/usr/lib/python2.7/dist-packages/urllib3/connection.py", line 237, in connect
    ssl_version=resolved_ssl_version)
  File "/usr/lib/python2.7/dist-packages/urllib3/util/ssl_.py", line 123, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python2.7/ssl.py", line 350, in wrap_socket
    _context=self)
TypeError: __init__() got an unexpected keyword argument 'server_hostname'
<Greenlet at 0xb65269dcL: <bound method AsyncRequest.send of <grequests.AsyncRequest object at 0xb64e508c>>(stream=False)> failed with TypeError
```

I am using a virtualenv on an up-to-date Debian/testing (jessie).

It seems that grequests redefined an ssl module, that want to have the same API as Python 2.7. Unfortunately, they introduced server_hostname and other niceties in Python 2.7.9 that seem to have been backported in Debian. And the wrapper is now broken.
Maybe I'm missing something, somewhere… But now, I am unable to have results for all HTTPS sites. This is quite annoying :smile:

**Comment by amitt001** [comment]
I think this problem still persists at least with python 2.7.9(atleast, on heroku).
A workaround is to degrade to python version 2.7.8 or lower.

**Comment by avelino** [comment]
Is expected when resolves in Python version 2.7.9?

**Comment by FuqiangZhu** [comment]
I'm sorry.
